5 compile warnings on MacOs

building on MacOS release mode results in the following 5 warnings:

1. warning: expression with side effects will be evaluated despite being used as an operand to 'typeid' [-Wpotentially-evaluated-expression]

```
[ 17%] Building CXX object lib/IlmCtl/CMakeFiles/IlmCtl.dir/CtlType.cpp.o
/Users/runner/work/CTL/CTL/lib/IlmCtl/CtlType.cpp:207:43: warning: expression with side effects will be evaluated despite being used as an operand to 'typeid' [-Wpotentially-evaluated-expression]
    return t && typeid (*this) == typeid (*t);
                                          ^
/Users/runner/work/CTL/CTL/lib/IlmCtl/CtlType.cpp:285:43: warning: expression with side effects will be evaluated despite being used as an operand to 'typeid' [-Wpotentially-evaluated-expression]
    return t && typeid (*this) == typeid (*t);
                                          ^
/Users/runner/work/CTL/CTL/lib/IlmCtl/CtlType.cpp:466:43: warning: expression with side effects will be evaluated despite being used as an operand to 'typeid' [-Wpotentially-evaluated-expression]
    return t && typeid (*this) == typeid (*t);
                                          ^
/Users/runner/work/CTL/CTL/lib/IlmCtl/CtlType.cpp:681:43: warning: expression with side effects will be evaluated despite being used as an operand to 'typeid' [-Wpotentially-evaluated-expression]
    return t && typeid (*this) == typeid (*t);
                                          ^
/Users/runner/work/CTL/CTL/lib/IlmCtl/CtlType.cpp:895:43: warning: expression with side effects will be evaluated despite being used as an operand to 'typeid' [-Wpotentially-evaluated-expression]
    return t && typeid (*this) == typeid (*t);
                                          ^
/Users/runner/work/CTL/CTL/lib/IlmCtl/CtlType.cpp:1079:43: warning: expression with side effects will be evaluated despite being used as an operand to 'typeid' [-Wpotentially-evaluated-expression]
    return t && typeid (*this) == typeid (*t);
                                          ^
/Users/runner/work/CTL/CTL/lib/IlmCtl/CtlType.cpp:1264:43: warning: expression with side effects will be evaluated despite being used as an operand to 'typeid' [-Wpotentially-evaluated-expression]
    return t && typeid (*this) == typeid (*t);
                                          ^
7 warnings generated.
```

2. warning: result of comparison of constant 'little_endian' (3) with expression of type 'bool' is always true [-Wtautological-constant-out-of-range-compare]

```
[ 51%] Building CXX object lib/dpx/CMakeFiles/ctldpx.dir/dpx.cc.o
/Users/runner/work/CTL/CTL/lib/dpx/dpx.cc:391:25: warning: result of comparison of constant 'little_endian' (3) with expression of type 'bool' is always true [-Wtautological-constant-out-of-range-compare]
        if(cpu_is_little_endian!=little_endian) {
           ~~~~~~~~~~~~~~~~~~~~^ ~~~~~~~~~~~~~
1 warning generated.
```

3. warning: self-comparison always evaluates to false [-Wtautological-compare]

```
[ 58%] Building CXX object lib/dpx/CMakeFiles/ctldpx.dir/dpx_rw.cc.o
/Users/runner/work/CTL/CTL/lib/dpx/dpx_rw.cc:264:27: warning: self-comparison always evaluates to false [-Wtautological-compare]
        if(h->current_endian_mode!=h->current_endian_mode) {
                                 ^
1 warning generated.
[ 59%] Linking CXX static library libctldpx.a
[ 59%] Built target ctldpx
```

4. warning: 'sprintf' is deprecated

```
[ 61%] Building CXX object ctlrender/CMakeFiles/ctlrender.dir/transform.cc.o
/Users/runner/work/CTL/CTL/ctlrender/transform.cc:397:4: warning: 'sprintf' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
                        sprintf( message_text, "CTL file must contain either a main or <module_name> (%s) function", module);
                        ^
/Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.1.sdk/usr/include/stdio.h:188:1: note: 'sprintf' has been explicitly marked deprecated here
__deprecated_msg("This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead.")
^
/Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.1.sdk/usr/include/sys/cdefs.h:215:48: note: expanded from macro '__deprecated_msg'
        #define __deprecated_msg(_msg) __attribute__((__deprecated__(_msg)))
                                                      ^
1 warning generated.
```

5. warning: 'uint32' is deprecated - the same warning as #138 
```

[ 67%] Building CXX object ctlrender/CMakeFiles/ctlrender.dir/tiff_file.cc.o
/Users/runner/work/CTL/CTL/ctlrender/tiff_file.cc:487:30: warning: 'uint32' is deprecated [-Wdeprecated-declarations]
        TIFFReadRGBAImage(t, w, h, (uint32 *)temp_buffer, 0);
                                    ^
/usr/local/include/tiff.h:86:45: note: 'uint32' has been explicitly marked deprecated here
typedef TIFF_MSC_DEPRECATED uint32_t uint32 TIFF_GCC_DEPRECATED;
                                            ^
/usr/local/include/tiff.h:66:44: note: expanded from macro 'TIFF_GCC_DEPRECATED'
#define TIFF_GCC_DEPRECATED __attribute__((deprecated))
                                           ^
1 warning generated.
```

--- a/ctlrender/transform.cc
+++ b/ctlrender/transform.cc
@@ -394,7 +394,7 @@ void run_ctl_transform(const ctl_operation_t &ctl_operation, CTLResults *ctl_res
             }
         } catch (...) {
 			char message_text[512] = {'\0'};
-			sprintf( message_text, "CTL file must contain either a main or <module_name> (%s) function", module);
+			snprintf( message_text, 512, "CTL file must contain either a main or <module_name> (%s) function", module);
             THROW(Iex::ArgExc, message_text);
         }		
 

--- a/lib/IlmCtl/CtlType.cpp
+++ b/lib/IlmCtl/CtlType.cpp
@@ -204,7 +204,10 @@ VoidType::VoidType (): DataType ()
 bool
 VoidType::isSameTypeAs (const TypePtr &t) const
 {
-    return t && typeid (*this) == typeid (*t);
+	const type_info& ti1 = typeid(*this);
+	const type_info& ti2 = typeid(*t);
+	return t && ti1 == ti2;
+    //return t && typeid (*this) == typeid (*t);
 }
 
 
@@ -282,7 +285,10 @@ BoolType::BoolType () : DataType ()
 bool
 BoolType::isSameTypeAs (const TypePtr &t) const
 {
-    return t && typeid (*this) == typeid (*t);
+    const type_info& ti1 = typeid(*this);
+	const type_info& ti2 = typeid(*t);
+	return t && ti1 == ti2;
+	//return t && typeid (*this) == typeid (*t);
 }
 
 
@@ -463,7 +469,10 @@ IntType::IntType () : DataType ()
 bool
 IntType::isSameTypeAs (const TypePtr &t) const
 {
-    return t && typeid (*this) == typeid (*t);
+    const type_info& ti1 = typeid(*this);
+	const type_info& ti2 = typeid(*t);
+	return t && ti1 == ti2;
+	//return t && typeid (*this) == typeid (*t);
 }
 
 
@@ -678,7 +687,10 @@ UIntType::UIntType () : DataType ()
 bool
 UIntType::isSameTypeAs (const TypePtr &t) const
 {
-    return t && typeid (*this) == typeid (*t);
+    const type_info& ti1 = typeid(*this);
+	const type_info& ti2 = typeid(*t);
+	return t && ti1 == ti2;
+	//return t && typeid (*this) == typeid (*t);
 }
 
 
@@ -892,7 +904,10 @@ HalfType::HalfType (): DataType ()
 bool
 HalfType::isSameTypeAs (const TypePtr &t) const
 {
-    return t && typeid (*this) == typeid (*t);
+    const type_info& ti1 = typeid(*this);
+	const type_info& ti2 = typeid(*t);
+	return t && ti1 == ti2;
+	//return t && typeid (*this) == typeid (*t);
 }
 
 
@@ -1076,7 +1091,10 @@ FloatType::FloatType (): DataType ()
 bool
 FloatType::isSameTypeAs (const TypePtr &t) const
 {
-    return t && typeid (*this) == typeid (*t);
+    const type_info& ti1 = typeid(*this);
+	const type_info& ti2 = typeid(*t);
+	return t && ti1 == ti2;
+	//return t && typeid (*this) == typeid (*t);
 }
 
 
@@ -1261,7 +1279,10 @@ StringType::StringType (): DataType ()
 bool
 StringType::isSameTypeAs (const TypePtr &t) const
 {
-    return t && typeid (*this) == typeid (*t);
+    const type_info& ti1 = typeid(*this);
+	const type_info& ti2 = typeid(*t);
+	return t && ti1 == ti2;
+	//return t && typeid (*this) == typeid (*t);
 }
 
 

--- a/lib/IlmCtl/CtlType.cpp
+++ b/lib/IlmCtl/CtlType.cpp
@@ -204,8 +204,10 @@ VoidType::VoidType (): DataType ()
 bool
 VoidType::isSameTypeAs (const TypePtr &t) const
 {
-	const type_info& ti1 = typeid(*this);
-	const type_info& ti2 = typeid(*t);
+	auto& t1 = *this;
+	const type_info& ti1 = typeid(t1);
+	auto& t2 = *t;
+	const type_info& ti2 = typeid(t2);
 	return t && ti1 == ti2;
     //return t && typeid (*this) == typeid (*t);
 }
@@ -285,8 +287,10 @@ BoolType::BoolType () : DataType ()
 bool
 BoolType::isSameTypeAs (const TypePtr &t) const
 {
-    const type_info& ti1 = typeid(*this);
-	const type_info& ti2 = typeid(*t);
+    auto& t1 = *this;
+	const type_info& ti1 = typeid(t1);
+	auto& t2 = *t;
+	const type_info& ti2 = typeid(t2);
 	return t && ti1 == ti2;
 	//return t && typeid (*this) == typeid (*t);
 }
@@ -469,8 +473,10 @@ IntType::IntType () : DataType ()
 bool
 IntType::isSameTypeAs (const TypePtr &t) const
 {
-    const type_info& ti1 = typeid(*this);
-	const type_info& ti2 = typeid(*t);
+    auto& t1 = *this;
+	const type_info& ti1 = typeid(t1);
+	auto& t2 = *t;
+	const type_info& ti2 = typeid(t2);
 	return t && ti1 == ti2;
 	//return t && typeid (*this) == typeid (*t);
 }
@@ -687,8 +693,10 @@ UIntType::UIntType () : DataType ()
 bool
 UIntType::isSameTypeAs (const TypePtr &t) const
 {
-    const type_info& ti1 = typeid(*this);
-	const type_info& ti2 = typeid(*t);
+    auto& t1 = *this;
+	const type_info& ti1 = typeid(t1);
+	auto& t2 = *t;
+	const type_info& ti2 = typeid(t2);
 	return t && ti1 == ti2;
 	//return t && typeid (*this) == typeid (*t);
 }
@@ -904,8 +912,10 @@ HalfType::HalfType (): DataType ()
 bool
 HalfType::isSameTypeAs (const TypePtr &t) const
 {
-    const type_info& ti1 = typeid(*this);
-	const type_info& ti2 = typeid(*t);
+    auto& t1 = *this;
+	const type_info& ti1 = typeid(t1);
+	auto& t2 = *t;
+	const type_info& ti2 = typeid(t2);
 	return t && ti1 == ti2;
 	//return t && typeid (*this) == typeid (*t);
 }
@@ -1091,8 +1101,10 @@ FloatType::FloatType (): DataType ()
 bool
 FloatType::isSameTypeAs (const TypePtr &t) const
 {
-    const type_info& ti1 = typeid(*this);
-	const type_info& ti2 = typeid(*t);
+    auto& t1 = *this;
+	const type_info& ti1 = typeid(t1);
+	auto& t2 = *t;
+	const type_info& ti2 = typeid(t2);
 	return t && ti1 == ti2;
 	//return t && typeid (*this) == typeid (*t);
 }
@@ -1279,8 +1291,10 @@ StringType::StringType (): DataType ()
 bool
 StringType::isSameTypeAs (const TypePtr &t) const
 {
-    const type_info& ti1 = typeid(*this);
-	const type_info& ti2 = typeid(*t);
+    auto& t1 = *this;
+	const type_info& ti1 = typeid(t1);
+	auto& t2 = *t;
+	const type_info& ti2 = typeid(t2);
 	return t && ti1 == ti2;
 	//return t && typeid (*this) == typeid (*t);
 }

--- a/lib/dpx/dpx.cc
+++ b/lib/dpx/dpx.cc
@@ -387,10 +387,11 @@ void dpx::write(std::ostream *os)
 		cpu_is_little_endian=TRUE;
 	}
 
-	_need_byteswap=FALSE;
-	if(cpu_is_little_endian!=little_endian) {
-		_need_byteswap=TRUE;
-	}
+	//_need_byteswap=FALSE;
+	//if(cpu_is_little_endian!=little_endian) {
+	//	_need_byteswap=TRUE;
+	//}
+	_need_byteswap=TRUE;
 
 	// struct tm source_creation_time_tm;
 	// time_t source_creation_time_time;

--- a/lib/dpx/dpx_rw.cc
+++ b/lib/dpx/dpx_rw.cc
@@ -261,7 +261,8 @@ void rwinfo::write_init(std::ostream *o, dpx *h) {
 	if(h->current_compliance!=h->compliance) {
 		// XXX changed validation level... This will probably end in tears...
 	}
-	if(h->current_endian_mode!=h->current_endian_mode) {
+	//if(h->current_endian_mode!=h->current_endian_mode) {
+	if(h->current_endian_mode!=h->endian_mode) {
 		// XXX byteswap changed. this will probably end in tears...
 	}
 }

